src/field_theory/subfield.lean

```diff
@@ -190,17 +190,28 @@ lemma gsmul_mem {x : K} (hx : x ∈ s) (n : ℤ) :
 lemma coe_int_mem (n : ℤ) : (n : K) ∈ s :=
 by simp only [← gsmul_one, gsmul_mem, one_mem]
 
+/- Unification is significantly faster if we define these using `.1` instead of `↑`. See gh-6514
+for an extreme example. It doesn't matter really anyway, as the simp lemmas below all use the
+coercion form. -/
+instance : has_zero s := ⟨⟨0, s.zero_mem⟩⟩
+instance : has_add s := ⟨λ x y, ⟨x.1 + y.1, s.add_mem x.2 y.2⟩⟩
+instance : has_sub s := ⟨λ x y, ⟨x.1 - y.1, s.sub_mem x.2 y.2⟩⟩
+instance : has_neg s := ⟨λ x, ⟨-x.1, s.neg_mem x.2⟩⟩
+instance : has_one s := ⟨⟨1, s.one_mem⟩⟩
+instance : has_mul s := ⟨λ x y, ⟨x.1 * y.1, s.mul_mem x.2 y.2⟩⟩
+instance : has_div s := ⟨λ x y, ⟨x.1 / y.1, s.div_mem x.2 y.2⟩⟩
+instance : has_inv s := ⟨λ x, ⟨x.1⁻¹, s.inv_mem x.2⟩⟩
+
 /-- A subfield inherits a field structure -/
 instance to_field : field s :=
-{ inv := λ x, ⟨x⁻¹, s.inv_mem x.2⟩,
-  inv_zero := subtype.ext inv_zero,
-  mul_inv_cancel := λ x hx, subtype.ext (mul_inv_cancel (mt s.to_subring.coe_eq_zero_iff.mp hx)),
-  exists_pair_ne := ⟨⟨0, s.zero_mem⟩, ⟨1, s.one_mem⟩, mt subtype.mk_eq_mk.mp zero_ne_one⟩,
-  ..s.to_subring.integral_domain }
+subtype.coe_injective.field coe
+  rfl rfl (λ _ _, rfl) (λ _ _, rfl) (λ _, rfl) (λ _ _, rfl) (λ _, rfl) (λ _ _, rfl)
```

Comment:
I'm not really sure how big a difference this makes. Certainly, this replaces three slightly non-trivial proofs with a bunch of extremely trivial ones. It also means that `/` on subfields is now defeq to the parent field, which was previously not the case.

The `.1` vs `↑` thing makes a massive difference in gh-6514, so I can only assume that something is up here.